db/schemaGenerator/data_generators.js

```diff
@@ -0,0 +1,88 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+var sentenceGenerator = (minNumOfWords, maxNumOfWords, possiblyComma) => {
+    var tempMem_chosenIndexes = [];
+    var sentence = '';
+    var numOfWords = randomNum(minNumOfWords, maxNumOfWords);
+    var placement;
+    if (possiblyComma) {
+        var comma = randomNum(0, 1);
+        placement = comma ? randomNum(0, numOfWords - 2) : undefined;
+    }
+       
+    for (var i = 0; i < numOfWords; i++) {  
+        do { var chosenIndex = randomNum(0, words.length - 1); } 
+        while (tempMem_chosenIndexes.indexOf(chosenIndex) > -1);
+
+        tempMem_chosenIndexes.push(chosenIndex);
+
+        sentence+= i === placement
+            ? words[chosenIndex] + ', ' 
+            : words[chosenIndex] + ' ';
+    }
+
+    tempMem_chosenIndexes = [];
+    sentence = sentence[0].toUpperCase() + sentence.slice(1, sentence.length - 1) + '. ';
+        
+    return sentence;
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+var paragraphGenerator = (numOfSentences, minNumWords, maxNumWords, possiblyComma) => {
+    var paragraph = '';
+    for(var i = 0; i < numOfSentences; i++) {
+        paragraph+= sentenceGenerator(minNumWords, maxNumWords, possiblyComma);
+    }
+    paragraph = paragraph.substring(0, paragraph.length - 1);
+    return paragraph;
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+module.exports.nikeIpsumGenerator = (num, minNumSent, maxNumSent, minNumWords, maxNumWords, possiblyComma) => {
+    var ipsums = [];
+    for(var i = 0; i < num; i++) {
+        var numOfSentences = randomNum(minNumSent, maxNumSent);
+        var paragraph = paragraphGenerator(numOfSentences, minNumWords, maxNumWords, possiblyComma);
+        
+        ipsums.push(paragraph);
+    }
+    return ipsums;
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+module.exports.insertGenerator = (numOfInserts, table, valueNames, ipsums) => {
+    var inserts = [];
+    
+    valueNames = `(${valueNames.join(', ')})`;
+
+    for (var i = 0; i < numOfInserts; i++) {
+        var values = [];
+        for (var j = 0; j < ipsums.length; j++) {
+            var value = ipsums[j][i];
+            if (typeof value === 'string') {
+                values.push(`"${ipsums[j][i]}"`);
+            } else {
+                values.push(`${ipsums[j][i]}`);
+            }
+            
+        }
+        values = `(${values.join(', ')})`;
+
+        inserts.push(`INSERT INTO ${table} ${valueNames} VALUES ${values};`);
+    }
+
+    return inserts;
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+module.exports.generateRandomAmountForEachProductId = (numOfProducts, minNumOfIds, maxNumOfIds) => {
```

Comment:
This is to generate an array with a random amount of product id's.

[1, 1, 1, 2, 2, 3, 3, 3, 4, 4, 4, 4.....etc.

Just so I could have an array to loop through. In order to create the benefits table and details table (both being many to 1 relationship with the products table).

Reply:
Ahhh - OK. Was a little hard to decipher from context, but I understand your intent now. Still interested in officially formalizing whether you're planning to change the associations of descriptions (etc) to products rather than styles from a structural change management standpoint AND selfishly because it determines whether my module needs to talk to yours or not. 

It might seem harder to do it based on styles, but since you're generating random data, you would just need to get the total number of styles generated from the seeding table (I'll make a final one this afternoon) and use that number instead of 100.